src/update_collection_metadata.py

```diff
@@ -4,25 +4,22 @@
 import github3
 import pendulum
 from prefect import flow, task
-from prefect.blocks.system import Secret
 from prefect.deployments import run_deployment
-from prefect.states import Completed
+from prefect.server.schemas.core import FlowRun
+from prefect.states import Completed, Failed, State
+from prefect.utilities.collections import listrepr
 
 import utils
 from generate_block_metadata import update_block_metadata_for_collection
 from generate_flow_metadata import update_flow_metadata_for_collection
 
 
-@task
-def collection_needs_update(collection_name: str, github_token_name: str) -> bool:
+async def collection_needs_update(collection_name: str) -> tuple[str, bool]:
     """
     Checks if the collection needs to be updated.
     """
-    github_token = Secret.load(github_token_name)
-    gh = github3.login(token=github_token.get())
-
-    collection_repo = gh.repository("PrefectHQ", collection_name)
-    registry_repo = gh.repository("PrefectHQ", "prefect-collection-registry")
+    collection_repo = await utils.get_repo(collection_name)
+    registry_repo = await utils.get_repo("prefect-collection-registry")
```

Comment:
Should we asyncio.gather this too?

Reply:
same as above, I can add this later but i don't see a big practical benefit in doing this now